### PR TITLE
SPARKC-181(remove duplicate)

### PR DESCRIPTION
**Pull request by ponkin**
I was trying to remove duplicate code from CassandraJoinRDD and CassandraLeftJoinRDD.

### CODE REVIEW

**Comment by ponkin** [comment]
@ostefano still not sure about the name of the trait - AbstractCassandraJoin seems to be meaningless.

File: spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/rdd/AbstractCassandraJoin.scala [hunk 1]
```diff
@@ -0,0 +1,190 @@
+package com.datastax.spark.connector.rdd
+
+import org.apache.spark.metrics.InputMetricsUpdater
+
+import com.datastax.spark.connector._
+import com.datastax.spark.connector.writer._
+import com.datastax.spark.connector.rdd.reader._
+import com.datastax.spark.connector.util.CqlWhereParser.{ EqPredicate, InListPredicate, InPredicate, RangePredicate }
+import com.datastax.spark.connector.util.{ CountingIterator, CqlWhereParser }
```
Comment (ostefano) [inline spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/rdd/AbstractCassandraJoin.scala]: Spaces here between imports and brackets do not seem consistent with the connector code base. I say we should try to be as consistent as possible. What do you think?

File: spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/rdd/CassandraJoinRDD.scala [hunk 2]
```diff
@@ -2,19 +2,19 @@ package com.datastax.spark.connector.rdd
 
 import org.apache.spark.metrics.InputMetricsUpdater
 
-import com.datastax.driver.core.{ResultSet, Session}
+import com.datastax.driver.core.{ ResultSet, Session }
 import com.datastax.spark.connector._
```
Comment (ostefano) [inline spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/rdd/CassandraJoinRDD.scala:4]: Same thing here

File: spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/rdd/CassandraJoinRDD.scala [hunk 3]
```diff
@@ -25,34 +25,37 @@ import com.google.common.util.concurrent.{FutureCallback, Futures, SettableFutur
...
+  override val manualRowWriter: Option[RowWriter[L]] = None
+)(
   implicit
-    val leftClassTag: ClassTag[L],
-    val rightClassTag: ClassTag[R],
-    @transient val rowWriterFactory: RowWriterFactory[L],
-    @transient val rowReaderFactory: RowReaderFactory[R])
-  extends CassandraRDD[(L, R)](left.sparkContext, left.dependencies)
-  with CassandraTableRowReaderProvider[R] {
+  val leftClassTag: ClassTag[L],
```
Comment (ostefano) [inline spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/rdd/CassandraJoinRDD.scala:37]: I think something is off re the identation.

**Comment by ostefano** [comment]
Cross check with Russel on Gitter/Irc regarding the naming. Re the pull request looks ok (left some tiny comments re formatting). Btw, did you check the tests are all successful? If so, just rebase it (and squash the commits into one), and I will accept it right away :)

**Pull request closed without merging**